Core/src/org/sleuthkit/autopsy/casemodule/Case.java

```diff
@@ -1468,15 +1471,104 @@ public String getOutputDirectory() {
+                // if no custom directory defined, use the system directory.
+                if (StringUtils.isBlank(customDirectory)) {
+                    return getSystemTempDirPrefix();
+                } else {
+                    // equivalent to [custom temp directory]/[app name]
+                    return new String[] { customDirectory, APP_NAME, getName() };
+                }
+            case SYSTEM:
+            default:
+                // default to the system directory.
+                return getSystemTempDirPrefix();
+        }
+    }
+    
+    /** 
+     * The sub folders of the temp directory to be used based on the case type.
+     * See JIRA-7505 for more information.
+     * @return The sub folders to be used in the temp directory based on the 
+     * case type.  Guaranteed to be non-null.
+     */
+    private String[] getTempDirCaseTypeSuffix() {
+        return (CaseType.MULTI_USER_CASE.equals(getCaseType())) 
+                // equivalent to [host]/Temp
+                ? new String[]{ NetworkUtils.getLocalHostName(), TEMP_NAME } 
+                // equivalent to Temp
+                : new String[]{ TEMP_NAME };
+    }
+    
+    
+
+    
     /**
      * Gets the full path to the temp directory for this case, creating it if it
-     * does not exist.
+     * does not exist.  
+     * 
+     * NOTE: UserMachinePreferences may also be affected by changes in 
+     * this method.  See JIRA-7505 for more information.
      *
      * @return The temp subdirectory path.
      */
     public String getTempDirectory() {
-        return UserMachinePreferences.getTempDirectory();
+        Stream<String> choicePrefix = Stream.of(getTempDirChoicePrefix());
+        Stream<String> caseTypeSuffix = Stream.of(getTempDirCaseTypeSuffix());
+        
+        // combine prefix based on temp directory choice and suffix based on case type.
+        List<String> pathPieces = Stream.concat(choicePrefix, caseTypeSuffix)
+            .collect(Collectors.toList());
+        
+        File dirFile;
+        if (pathPieces.isEmpty()) {
+            // in the event of a severe failure that should never happen
+            // because both prefix and suffix calls should be non-empty, 
+            // fallback to a subdirectory of java tmpdir
+            logger.log(Level.SEVERE, "No path pieces provided for temp directory.  Falling back to java.io.tmpdir.");
+            dirFile = Paths.get(System.getProperty("java.io.tmpdir"), APP_NAME).toFile();
+        } else {
+            String first = pathPieces.get(0);
+            List<String> remainingList = pathPieces.subList(1, pathPieces.size());
+            dirFile = Paths.get(first, remainingList.toArray(new String[remainingList.size()])).toFile();
+        }
+        
+        // ensure directory exists
+        if (!dirFile.exists()) {
+            dirFile.mkdirs();
+        }
+        
+        return dirFile.getAbsolutePath();
```

Comment:
I think that a somewhat different algorithm will allow us to separate concerns a little bit more. I think the key is to think of the application temp directory and the temp directory choice as two separate things. If there is no current case, the client can call Path UserMachinePreferences.getAppTempDirPath(); otherwise most clients can call Case.getTempDirectory(). 

We can also use the Path and Paths classes to work at a higher level of abstraction than String arrays and perhaps make the code easier to follow.  Here is what I'm thinking can go here:

```
if UserMachinePreferences.getTempDirectoryType() == CASE
	if getCaseType() == SINGLE_USER_CASE
		return Paths.get(getCaseDirectory(), TEMP_FOLDER).toString()
	else
		return Paths.get(getCaseDirectory(), getHostName(), TEMP_FOLDER).toString()
else
	Path appTempDirPath = UserMachinePreferences.getAppTempDirPath()
	Path caseTempDirPath = appTempDirPath.subpath(0, appTempDirPath.getNameCount() - 1) 
	caseTempDirPath = Paths.get(caseTempDirPath, getName());
	if getCaseType() == SINGLE_USER_CASE
		return Paths.get(caseTempDirPath.toString(), TEMP_FOLDER).toString()
	else
		return Paths.get(caseTempDirPath.toString(), getHostName(), TEMP_FOLDER)
```